pkg/catalog/catalog.go

```diff
@@ -235,8 +236,12 @@ type Config struct {
 	PathProvider          *upload.PathPartitionProvider
 	// ConflictResolvers alternative conflict resolvers (if nil, the default behavior is kept)
 	ConflictResolvers []graveler.ConflictResolver
+	ApiErrorCallback  ApiErrorCallback
 }
 
+// ApiErrorCallback is a callback function used to classify errors and convert them to status codes and error messages.
+type ApiErrorCallback func(logger logging.Logger, err error) (int, string, bool)
```

Comment:
It's not a callback anymore it's just a conversion function i.e `errToApiErr` or something like this. 
Callback is a bit confusing it makes the sense that it has some side effect but it really just regular helper function.
The comment about naming of this type and the field inside Catalog.